app/controllers/tea_times_controller.rb

```diff
@@ -1,13 +1,38 @@
 class TeaTimesController < ApplicationController
   helper TeaTimesHelper
-  before_action :set_tea_time, except: [:index, :new, :create]
-  before_action :authenticate_user!
-  before_action :authorized?, only: [:new, :edit, :create, :update, :cancel, :destroy, :index]
+  before_action :set_tea_time, except: [:index, :list, :new, :create]
+  before_action :authenticate_user!, except: [:index]
+  before_action :authorized?, only: [:new, :edit, :create, :update, :cancel, :destroy, :list]
   before_action :use_new_styles, except: [:create, :update, :cancel, :destroy]
 
   # GET /tea_times
   # GET /tea_times.json
   def index
+    @tea_times = TeaTime.this_month.order(start_time: :asc).includes(:city).all
+
```

Comment:
Here we are converting a list of tea times into a hash like:

```
{
  "San Francisco": [TeaTime, TeaTime, ...]
  "NYC": [TeaTime, TeaTime, ...]
}
```

along with a list of names in `@cities`.

Reply:
We don't have these AR scopes, so you'll need something like `where(:created_at => Time.now.beginning_of_month..Time.now.end_of_month)` instead.

It's also going to give you an array of `TeaTime`. If you want it grouped by city you'll have to go from that model instead.

Reply:
This will return an array of `TeaTime`, not a hash. Additionally, `this_month` isn't a standard AR Scope and it's not defined on the TT model, so you'll need something like `where(:created_at => Time.now.beginning_of_month..Time.now.end_of_month)` instead I think. Might want to make the span a rolling 4 week period too so it doesn't weirdly cut over towards the end of a month?

Reply:
@nickbarnwell 

Re: array: Whoops, my bad for not being clear. I meant this comment to 'introduce' the following lines of code which are doing the process of converting that list to a hash.

Re: scope: I added a custom scope in the model, let me know if it looks right: https://github.com/TeaWithStrangers/tws-on-rails/pull/736/files#diff-c0448cf921e60c620dcf5c85c1718605R35

Re: rolling period: This was something Ankit requested in the spec - I think the idea was to have tea times come out in "batches" at the beginning of the month.

Reply:
@markbao @nickbarnwell 

Re: rolling period — Mark is right on what I suggested in the spec. We discussed having following month's tea times be available with the click of a "Show Next Month's Tea Times" button. Not sure if we landed on a conclusion there. @markbao Do you remember?